### PR TITLE
Search project state metadata

### PR DESCRIPTION
**Pull request by cdolm92**
## What

@kickstarter/native-squad 
Added metadata to projects in search view. 

Before:

  screen shot 2017-01-24 at 4 49 48 pm

After:

  screen shot 2017-01-24 at 4 44 21 pm

### CODE REVIEW

**Comment by cdolm92** [comment]
waiting on a string for this

File: Library/ViewModels/MostPopularSearchProjectCellViewModel.swift [hunk 1]
```diff
@@ -8,8 +8,10 @@ public protocol MostPopularSearchProjectCellViewModelInputs {
 }
 
 public protocol MostPopularSearchProjectCellViewModelOutputs {
-  var fundingLabelText: Signal<String, NoError> { get }
-  var fundingProgress: Signal<Float, NoError> { get }
+  var deadlineSubtitleLabelText: Signal<String, NoError> { get }
+  var deadlineTitleLabelText: Signal<String, NoError> { get }
+  var fundingSubtitleLabelText: Signal<String, NoError> { get }
+  var fundingTitleLabelText: Signal<String, NoError> { get }
```
Comment (mbrandonw) [inline Library/ViewModels/MostPopularSearchProjectCellViewModel.swift]: i see that you've split up the styling of the labels into two parts. that may not always be the best choice with localization (espesh thinking of japan coming soon!), so a more future-proof way would be to use simple html. that requires using our custom `SimpleHTMLLabel` class and having a new string.

File: Library/ViewModels/SearchProjectCellViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import KsApi
+import Prelude
+import ReactiveSwift
+import Result
+
+public protocol SearchProjectCellViewModelInputs {
```
Comment (mbrandonw) [inline Library/ViewModels/SearchProjectCellViewModel.swift]: the cells for the "most popular" cell and the other cells seem to have the same data now, so I think  we could just use a single view model to service them both.

File: Library/ViewModels/MostPopularSearchProjectCellViewModel.swift (hunk 1 above)
Comment (mbrandonw) [inline Library/ViewModels/MostPopularSearchProjectCellViewModel.swift]: also looks like we are missing some tests on these new outputs.

File: Kickstarter-iOS/Views/Cells/SearchProjectCell.swift [hunk 3]
```diff
@@ -65,7 +76,29 @@ internal final class SearchProjectCell: UITableViewCell, ValueCell {
...
+    self.viewModel.outputs.projectImageUrl
+      .observeForUI()
+      .on(event: { [weak self] _ in
+        self?.projectImageView.image = nil
+        self?.projectImageView.af_cancelImageRequest()
+      })
+      .skipNil()
+      .observeValues { [weak self] in
+        self?.projectImageView.af_setImage(withURL: $0)
+    }
```
Comment (mbrandonw) [inline Kickstarter-iOS/Views/Cells/SearchProjectCell.swift]: we can use the new `imageView.rac.imageUrl` reactive extension here if you are interested...

File: Kickstarter-iOS/Views/Cells/MostPopularCell.swift [hunk 4]
```diff
@@ -23,6 +23,6 @@ internal final class MostPopularCell: UITableViewCell, ValueCell {
     _ = self.mostPopularLabel
       |> UILabel.lens.font .~ .ksr_title1(size: 22)
       |> UILabel.lens.textColor .~ .ksr_text_navy_700
-      |> UILabel.lens.text %~ { _ in Strings.search_most_popular() }
+      |> UILabel.lens.text %~ { _ in "Popular Projects"}
```
Comment (mbrandonw) [inline Kickstarter-iOS/Views/Cells/MostPopularCell.swift]: let's wrap this in `localizedString` while we wait for the strings. that way if we `make strings` later and forget to update this we'll still get the translations.

**Pull request merged**